LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -10,69 +10,49 @@ import HealthKit
+    let initialValue: InsulinModelSettings
+    @State var value: InsulinModelSettings
+    let insulinSensitivitySchedule: InsulinSensitivitySchedule
+    let glucoseUnit: HKUnit
+    let supportedModelSettings: SupportedInsulinModelSettings
+    let mode: PresentationMode
+    let appName: String
+    let save: ((InsulinModelSettings) -> Void)?
 
     static let defaultInsulinSensitivitySchedule = InsulinSensitivitySchedule(unit: .milligramsPerDeciliter, dailyItems: [RepeatingScheduleValue<Double>(startTime: 0, value: 40)])!
-
+    
     static let defaultWalshInsulinModelDuration = TimeInterval(hours: 6)
-    static let validWalshModelDurationRange = InsulinModelSettings.validWalshModelDurationRange
 
-    var walshActionDuration: TimeInterval {
-        get {
-            if case .walsh(let walshModel) = insulinModelSettings {
-                return walshModel.actionDuration
-            } else {
-                return Self.defaultWalshInsulinModelDuration
+    var walshActionDuration: Binding<TimeInterval> {
+        Binding(
+            get: {
+                if case .walsh(let walshModel) = self.value {
+                    return walshModel.actionDuration
+                } else {
+                    return Self.defaultWalshInsulinModelDuration
+                }
+            },
+            set: { newValue in
+                precondition(InsulinModelSettings.validWalshModelDurationRange.contains(newValue))
+                self.value = .walsh(WalshInsulinModel(actionDuration: newValue))
             }
-        }
-        set {
-            precondition(Self.validWalshModelDurationRange.contains(newValue))
-            insulinModelSettings = .walsh(WalshInsulinModel(actionDuration: newValue))
-        }
-    }
-
-    public init(insulinModelSettings: InsulinModelSettings, insulinSensitivitySchedule: InsulinSensitivitySchedule?) {
-        self._insulinModelSettings = Published(wrappedValue: insulinModelSettings)
-        self.insulinSensitivitySchedule = insulinSensitivitySchedule ?? Self.defaultInsulinSensitivitySchedule
+        )
     }
-}
-
-public struct InsulinModelSelection: View, HorizontalSizeClassOverride {
-
-    let initialInsulinModelSettings: InsulinModelSettings
-    @ObservedObject var viewModel: InsulinModelSelectionViewModel
-    var glucoseUnit: HKUnit
-    var supportedModelSettings: SupportedInsulinModelSettings
-    let mode: PresentationMode
-    let appName: String
-    let onSave: ((InsulinModelSettings) -> Void)?
 
     public init(
-        viewModel: InsulinModelSelectionViewModel,
+        value: InsulinModelSettings,
+        insulinSensitivitySchedule: InsulinSensitivitySchedule?,
         glucoseUnit: HKUnit,
         supportedModelSettings: SupportedInsulinModelSettings,
         appName: String,
         mode: PresentationMode,
-        onSave: ((InsulinModelSettings) -> Void)? = nil
+        onSave save: ((InsulinModelSettings) -> Void)? = nil
```

Comment:
Seems this should be a required argument now, since it's the only way to persist changes

Reply:
Fair enough 👍